### PR TITLE
Fixed ampersand issue for offer rules

### PR DESCRIPTION
**Pull request by StanislavFedorov**
https://github.com/BroadleafCommerce/QA/issues/4663

### CODE REVIEW

**Comment by riteshadhikari17** [comment]
There was something wrong with build so I have kicked off another build manually. Also could you revert the wild card import?

**Comment by StanislavFedorov** [comment]
reverted import

File: core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/offer/service/processor/AbstractBaseProcessor.java [hunk 1]
```diff
@@ -279,6 +283,18 @@ protected boolean couldOrderItemMeetOfferRequirement(OfferItemCriteria criteria,
 
         return appliesToItem;
     }
+
```
Comment (danielcolgrove) [inline core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/offer/service/processor/AbstractBaseProcessor.java:282]: Would it make sense to pull this out into a helper class?  Are there other places we could use this logic.  If offer rules have this issue, what about Customer Segment, Targeters, other rule builders?

**Comment by StanislavFedorov** [comment]
>Would it make sense to pull this out into a helper class?

This sounds reasonable. I overlooked this.
The current solution is to move this "escapeSpecialCharacters" logic to the general class for all rule builders.
The changes looked pretty simple but I had to test all cases and implement 'unescapeSpecialCharacters'. So it took more time..
Because now we store that Ampersand in DB as &amp; , but before this, we just did the conversion for Offer rules "on the fly"
I checked this for Offers rules and for Customer Segment/Targeters

**Review by riteshadhikari17 — APPROVED** [review]

**Pull request merged**